app.js

```diff
@@ -46,14 +35,7 @@ var configDB = require('./config/database.js');
 // in your own promise library instead: http://mongoosejs.com/docs/promises.html
 mongoose.Promise = global.Promise;
 mongoose.connection.on('error', configDB.check);
-// Connect to proper db
-if (process.env.NODE_ENV == 'production') {
-  mongoose.connect(app.data.mongo_url.prod)
-} else if (process.env.NODE_ENV == 'testing'){
-  mongoose.connect(app.data.mongo_url.test)
-} else {
-  mongoose.connect(app.data.mongo_url.dev)
-}
+mongoose.connect(config.mongodb.link);
```

Comment:
Please don't remove existing functionality unless you replace it with some other method. In the current configuration, running tests requires a separate DB. NODE_ENV var determines the DB to which the app connects.

Also try not to squeeze all new features in a single PR, since it is hard to review and other cannot benefit from the features until the PR is merged (e.g. I cannot apply new lint rules for new code changes).

Let me know when this is ready for review.